packages/astro/src/content/utils.ts

```diff
@@ -167,6 +167,50 @@ export function getEntryConfigByExtMap<TEntryType extends ContentEntryType | Dat
 	return map;
 }
 
+export async function getSymlinkedContentCollections(
+	contentDir: URL
+): Promise<Map<string, string>> {
+	const contentPaths = new Map<string, string>();
+	const contentDirPath = fileURLToPath(contentDir);
+
+	if (!fsMod.existsSync(contentDirPath) || !fsMod.lstatSync(contentDirPath).isDirectory()) {
+		return contentPaths;
+	}
+
+	const contentDirEntries = await fsMod.promises.readdir(contentDir, { withFileTypes: true });
+	for (const entry of contentDirEntries) {
+		if (entry.isSymbolicLink()) {
+			const entryPath = path.join(contentDirPath, entry.name);
+			const realPath = await fsMod.promises.realpath(entryPath);
+			contentPaths.set(normalizePath(realPath), entry.name);
+		}
+	}
```

Comment:
nit: file system operations can come with runtime errors (permissions, etc.), I tend to wrap these operations in a `try/catch` to be more safe

Reply:
I'd lean towards silently ignoring them here, because we're just looking for symlinks and a failure probably means it's not a symlink.

Reply:
It really depends on the use case, really. In this case we could just ignore the error, and maybe log and error/warning using the logger (you should have access to it).

Reply:
Where would I find a logger instance? I'm calling this in [vite-plugin-content-imports](https://github.com/withastro/astro/blob/ab8746eea7dcbfadd7210c2fbfcb8b7dfca0ac3c/packages/astro/src/content/vite-plugin-content-imports.ts).

Reply:
It seems we don't pass it down to this plugin (the logger arrived *after* this plugin was created). You'll have to pass it like in this case:

https://github.com/withastro/astro/blob/2851b0aa2e2abe80ea603b53c67770e94980a8d3/packages/astro/src/core/create-vite.ts#L149

Reply:
Aha, thanks. I just pushed an update with error handling and renamed funciton, but no logging. I'll add the logs.

Reply:
OK, I've added logging now